crates/lib3h/src/engine/ghost_engine_wrapper.rs

```diff
@@ -0,0 +1,323 @@
+use crate::engine::ghost_engine::{CoreRequestContext, GhostEngineParentWrapper};
+use detach::Detach;
+use lib3h_ghost_actor::*;
+use lib3h_protocol::{
+    data_types::GenericResultData, error::Lib3hProtocolResult, protocol::*, protocol_client::*,
+    protocol_server::*, DidWork,
+};
+
+/// A wrapper for talking to lib3h using the legacy Lib3hClient/Server enums
+#[allow(dead_code)]
+struct LegacyLib3h<Engine, EngineError: 'static>
+where
+    Engine: GhostActor<
+        Lib3hToClient,
+        Lib3hToClientResponse,
+        ClientToLib3h,
+        ClientToLib3hResponse,
+        EngineError,
+    >,
+{
+    lib3h: Detach<GhostEngineParentWrapper<LegacyLib3h<Engine, EngineError>, Engine, EngineError>>,
+    #[allow(dead_code)]
+    name: String,
+    client_request_responses: Vec<Lib3hServerProtocol>,
+}
+
+fn server_failure(err: String, context: CoreRequestContext) -> Lib3hServerProtocol {
+    let failure_data = GenericResultData {
+        request_id: context.get_request_id(),
+        space_address: "space_addr".into(),
```

Comment:
I believe it would be nice to add `space_address` to  `CoreRequestContext`

Reply:
Do we allways have it?  For eample in ConnectData which is the space address?